### PR TITLE
Added deployment of development pages

### LINKED ISSUE #3: [Management] Set up GitHub actions for develop and master branches
- [x] Enable HTML build tests on PRs to develop branch
- [x] Enable HTML deployment for develop branch
- [x] Enable HTML deployment for master branch

### PR DESCRIPTION
**Pull request by sfarrens**
- Renamed `deploy_build.yml` to `deploy_pages.yml` to avoid confusion
- Created `deploy_dev.yml` to deploy development pages to `dev-pages` branch
  - Requires existing submodule in `gh-pages` pointing to `dev-pages` and named `develop`
  - Action automatically updates submodule to point to most recent commit
- Resolves #3

### CODE REVIEW

File: .github/workflows/deploy_dev.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          ref: gh-pages
+      - run: |
+          git submodule init
+          git submodule update --remote --merge
+          git submodule status
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add develop
+          git commit -m "updated dev-pages submodule"
+          git push origin gh-pages
```
Comment (aboucaud) [inline .github/workflows/deploy_dev.yml:49]: Why is that necessary ?

I imagine that if we remove the submodule as per #14, this will no longer be used.
Or maybe I just dont understand submodules 😄

Comment (sfarrens) [reply .github/workflows/deploy_dev.yml:49]: Right, I probably didn't make this very clear in the PR summary. 

- #14 is to get rid of the redundant submodule in the `master` branch
- I have created a new submodule in the `gh-pages` branch that is required for this PR
- The submodule is essentially a directory in `gh-pages` that is tied to a specific commit state of `dev-pages`. This way when you view https://cosmostat.github.io/Shear-and-PSF-Reading-Group/ you see the HTML files in `gh-pages` and when you view https://cosmostat.github.io/Shear-and-PSF-Reading-Group/develop you see the HTML files in the submodule directory
- In the GH action I first build the HTML from the `develop` branch, push this to the `dev-pages` branch and then finally update the commit state of the submodule in `gh-pages` so that the HTML is always up to date

Comment (aboucaud) [reply .github/workflows/deploy_dev.yml:49]: Alright I understand the need now, thanks ! 
As for the submodule workflow itself, it still isn't clear to me. I'll probably ask you a bit more details at the telecon.

**Review by aboucaud — APPROVED** [review]
Then that looks ok to me.

File: .github/workflows/deploy_dev.yml (hunk 1 above)
Comment (sfarrens) [reply .github/workflows/deploy_dev.yml:49]: Sure, no problem :)

**Comment by sfarrens** [comment]
@EiffL Let me know if you are also happy with the proposed set-up before I merge this.

**Review by EiffL — APPROVED** [review]
LGTM yeah
